src/sphinx_tags/__init__.py

```diff
@@ -294,24 +281,18 @@ def assign_entries(app):
     pages = []
     tags = {}
 
-    # Get document paths in the project that match specified file extensions
-    doc_paths = get_matching_files(
-        app.srcdir,
-        include_patterns=[f"**.{extension}" for extension in app.config.tags_extension],
-        exclude_patterns=app.config.exclude_patterns,
-    )
-
-    for path in doc_paths:
-        entry = Entry(Path(app.srcdir) / path)
+    for docname in app.env.found_docs:
```

Comment:
I had no idea this was a thing! Good find. I was looking for something like that for #58 but somehow missed it.